numpy/_core/src/umath/loops_arithm_fp.dispatch.cpp

```diff
@@ -0,0 +1,766 @@
+        ) {
+            T *rl_im = (T *)b_src0;
+            T rr, ri;
+            if constexpr (std::is_same_v<T, float>){
+                CFLOAT_pairwise_sum(&rr, &ri, b_src1, len * 2, b_ssrc1 / 2);
+            } else {
+                CDOUBLE_pairwise_sum(&rr, &ri, b_src1, len * 2, b_ssrc1 / 2);
+            }
+            rl_im[0] = op_func(rl_im[0], rr);
+            rl_im[1] = op_func(rl_im[1], ri);
+            return;
+        }
+    }
+
+#if NPY_HWY
+    if constexpr (kSupportLane<T>) {
+        // Certain versions of Apple clang (commonly used in CI images) produce
+        // non-deterministic output in the mul path with AVX2 enabled on x86_64.
+        // Work around by scalarising.
+        #if defined(NPY_CPU_AMD64) && defined(__clang__) \
+                && defined(__apple_build_version__) \
+                && __apple_build_version__ >= 14000000 \
+                && __apple_build_version__ < 14030000
+            if constexpr (Op::is_mul) {
+                goto loop_scalar;
+            }
+        #endif  // end affected Apple clang.
+
+        if (!is_mem_overlap(b_src0, b_ssrc0, b_dst, b_sdst, len) &&
+            !is_mem_overlap(b_src1, b_ssrc1, b_dst, b_sdst, len) &&
+            sizeof(T) == alignof(T) && b_ssrc0 % sizeof(T) == 0 && b_ssrc1 % sizeof(T) == 0 && 
+            b_sdst % sizeof(T) == 0 && b_sdst != 0) {
+            const T *src0 = (T*)b_src0;
+            const T *src1 = (T*)b_src1;
+                  T *dst  = (T*)b_dst;
+
+            const npy_intp ssrc0 = b_ssrc0 / sizeof(T);
+            const npy_intp ssrc1 = b_ssrc1 / sizeof(T);
+            const npy_intp sdst  = b_sdst  / sizeof(T);
+
+            HWY_LANES_CONSTEXPR int vstep = Lanes<T>();
+            const int wstep = vstep * 2;
+            const int hstep = vstep / 2;
+
+            // lots of specializations, to squeeze out max performance
+            // contiguous
+            if (ssrc0 == 2 && ssrc0 == ssrc1 && ssrc0 == sdst) {
+                for (; len >= vstep; len -= vstep, src0 += wstep, src1 += wstep, dst += wstep) {
+                    auto a0 = LoadU<T>(src0);
+                    auto a1 = LoadU<T>(src0 + vstep);
+                    auto b0 = LoadU<T>(src1);
+                    auto b1 = LoadU<T>(src1 + vstep);
+                    auto r0 = op_func(a0, b0);
+                    auto r1 = op_func(a1, b1);
+                    StoreU(r0, dst);
+                    StoreU(r1, dst + vstep);
+                }
+                for (; len > 0; len -= hstep, src0 += vstep, src1 += vstep, dst += vstep) {
+                    auto a = hn::LoadN(_Tag<T>(), src0, len*2);
+                    auto b = hn::LoadN(_Tag<T>(), src1, len*2);
+                    auto r = op_func(a, b);
+                    hn::StoreN(r, _Tag<T>(), dst, len*2);
+                }
+            }
+            // scalar 0
+            else if (ssrc0 == 0) {
+                auto a = hn::OddEven(Set<T>(src0[1]), Set<T>(src0[0]));
```

Comment:
```suggestion
                V a = hn::OddEven(Set(src0[1]), Set(src0[0]));
```
Same reasoning as `LoadU`.